src/Raven.Client/Documents/Operations/ETL/Queue/AzureQueueStorageConnectionSettings.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Linq;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.ETL.Queue;
+
+public sealed class AzureQueueStorageConnectionSettings
+{
+    public Authentication Authentication;
+
+    public string GetStorageUrl()
+    {
+        string storageAccountName = GetStorageAccountName();
+        return $"https://{storageAccountName}.queue.core.windows.net/";
+    }
+
+    public string GetStorageAccountName()
+    {
+        string storageAccountName = "";
+
+        if (Authentication.ConnectionString != null)
+        {
+            var accountNamePart = Authentication.ConnectionString.Split(';')
+                .FirstOrDefault(part => part.StartsWith("AccountName=", StringComparison.OrdinalIgnoreCase));
+
+            if (accountNamePart == null)
+            {
+                throw new ArgumentException("Storage account name not found in the connection string.",
+                    nameof(Authentication.ConnectionString));
+            }
+
+            storageAccountName = accountNamePart.Substring("AccountName=".Length);
+        }
+        else if (Authentication.EntraId != null)
+        {
+            storageAccountName = Authentication.EntraId.StorageAccountName;
+        }
+
+        return storageAccountName;
+    }
+
+    public DynamicJsonValue ToJson()
+    {
+        var json = new DynamicJsonValue
+        {
+            [nameof(Authentication)] = Authentication == null
+                ? null
+                : new DynamicJsonValue
+                {
+                    [nameof(Authentication.ConnectionString)] = Authentication.ConnectionString,
+                    [nameof(Authentication.Passwordless)] = Authentication.Passwordless,
+                    [nameof(Authentication.EntraId)] =
+                        Authentication.EntraId == null
+                            ? null
+                            : new DynamicJsonValue
+                            {
+                                [nameof(Authentication.EntraId.StorageAccountName)] =
+                                    Authentication?.EntraId?.StorageAccountName,
+                                [nameof(Authentication.EntraId.TenantId)] =
+                                    Authentication?.EntraId?.TenantId,
+                                [nameof(Authentication.EntraId.ClientId)] =
+                                    Authentication?.EntraId?.ClientId,
+                                [nameof(Authentication.EntraId.ClientSecret)] =
+                                    Authentication?.EntraId?.ClientSecret
+                            }
+                }
+        };
+
+        return json;
+    }
+}
+
+public sealed class Authentication
+{
+    public EntraId EntraId { get; set; }
+    public string ConnectionString { get; set; }
+    public bool Passwordless { get; set; }
```

Comment:
Why those aren't directly part of `AzureQueueStorageConnectionSettings`? Do we need `AzureQueueStorageConnectionSettings.Authentication` field?

Reply:
done
(I followed how we did for Elasticsearch, but maybe there also should be directly part of connection settings).

Reply:
In case of  `ElasticSearchConnectionString` it was a bit different because we provide node's URLs separately and then the authentication. Here we don't have such separation